bitsandbytes/backends/cpu.py

```diff
@@ -0,0 +1,287 @@
+import torch
+
+
+Tensor = torch.Tensor
+
+
+def assert_on_cpu(tensors):
+    on_cpu = True
+    for t in tensors:
+        if t is None: continue # NULL pointers are fine
+        on_cpu &= (t.device.type == 'cpu')
+    if not on_cpu:
+        raise TypeError(
+            'All input tensors need to be on CPU, but found some tensors to not be on CPU:\n' \
+            f' {[(t.shape, t.device) if isinstance(t, Tensor) else None for t in tensors]}'
+        )
+    return on_cpu
+
+
+@torch.compile(dynamic=True, options={"fx_graph_cache": True})
```

Comment:
This would require torch > 2.0, which is ok if XPU backend needs that, but I'm not sure there's reason enough to make it a hard requirement for all.

Reply:
Perhaps we can wrap a decorator to conditionally enable this.

Reply:
Thanks. Updated.